opl/patterns.py

```diff
@@ -24,5 +24,27 @@
                 LiteralString.value(expr, value);
             }
         ''',
+
+        # returns all artifact relationships
+        'predicateTarget': '''
+            (element : Class, target : Class, predicate : Property, predicateName : String, elementName : String, targetName : String) {
+                Class.name(element, elementName);
+                Class.ownedAttribute(element, predicate);
+                Property.name(predicate, predicateName);
+                Property.type(predicate, target);              
+                Class.name(target, targetName);
+            }
+        ''',
+
+        # get all literal string property values of an artifact
+        'propertyValue': '''
+            (artifact: Class, artifactName: String, property : Property, propertyName: String, value: String) {
+                Class.name(artifact, artifactName);             
+                Class.ownedAttribute(artifact, property);
+                Property.name(property, propertyName);
+                Property.defaultValue(property, propValue);    
+                LiteralString.value(propValue, value);
+            }
+        '''
```

Comment:
predicateTarget for getting all artifact relationships and propertyValue for getting all literal string property values